db/migrate/317_strip_keywords_terms.rb

```diff
@@ -0,0 +1,48 @@
+class StripKeywordsTerms < ActiveRecord::Migration
+
+  class ::MigrationKeyword < ActiveRecord::Base
+    self.table_name = 'keywords'
+  end
+
+  class ::MigrationMetaDatumKeyword < ActiveRecord::Base
+    self.table_name = 'meta_data_keywords'
+  end
+
+  def change
+    ActiveRecord::Base.transaction do
+      execute "SET session_replication_role = REPLICA"
+
+      ::MigrationKeyword.all.each do |keyword|
+        if keyword.term =~ Madek::Constants::TRIM_WHITESPACE_REGEXP
+          new_term = keyword.term.gsub(Madek::Constants::TRIM_WHITESPACE_REGEXP, '')
+
+          # if there is already another keyword for the same meta_key and stripped term
+          if other_keyword = ::MigrationKeyword.find_by(meta_key_id: keyword.meta_key_id, term: new_term)
+            # change all rows in meta_data_keywords to point keyword_id to the other keyword
+            ::MigrationMetaDatumKeyword.where(keyword_id: keyword.id).each do |md|
+              # make a copy of the meta_datum
+              new_md = md.dup
+              # destroy the meta_datum for the stripped keyword
+              md.destroy
+              # point keyword_id for the new meta_datum to the other keyword
+              new_md.keyword_id = other_keyword.id
+              # create new meta_datum_keyword only if there is not one already for the same keyword and meta_datum
+              unless ::MigrationMetaDatumKeyword.find_by(keyword_id: new_md.keyword_id, meta_datum_id: new_md.meta_datum_id)
+                new_md.save
```

Comment:
please make sure that the timestamps stay the same… (maybe we even have to "trick" PG, or AR can help…)